### PR TITLE
fix: reset text input fields when clear button is clicked

### LINKED ISSUE #858: Bug: resetting an input is not working as expected
### Describe the bug

When you fill an input and reset it using the `red X` button and focus on another input, the cleared value comes back (it will make sense after watching the loom below)
> **Note** I only tested it in `hub/insights/new` page
>
https://www.loom.com/share/d9f56da22fe64c828527e14cb4bd868d

### Steps to reproduce

1. Go to insights.opensauced.pizza
2. Create a new insight page
3. Add a page name, reset it with the close button and focus the add repo input
4. Notice some weird things happening

### Affected services

insights.opensauced.pizza

### Platforms

Desktop

### Browsers

Firefox, Other (add additional context)

### Environment

Production, Development

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by TAKANOME-DEV**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR fixes the input reset bug by clearing the input state when the clear (`x`) button is clicked
## Related Tickets & Documents

Fixes #858
## Mobile & Desktop Screenshots/Recordings

[insights-bug.webm](https://user-images.githubusercontent.com/79809121/218312324-38a8e5b2-19e7-46ab-885a-b7bd9a33691e.webm)

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by Deadreyo** [comment]
While this solution is correct, it adds an overhead of assigning a new prop `onReset` each place the component is needed. I wonder if you could remove that overhead by having the reset functionality be an internal part of the component, maybe by manually triggering an event with the value being an empty string?

**Comment by TAKANOME-DEV** [comment]
> While this solution is correct, it adds an overhead of assigning a new prop onReset each place the component is needed.
>
I agree.
> I wonder if you could remove that overhead by having the reset functionality be an internal part of the component, maybe by manually triggering an event with the value being an empty string?
>
Hum, would it be different to pass a handler as props to clear the input instead of using `onReset` handler?

Using `useRef` instead of `useState` will solve the problem but the close button won't show up. Do you think I should go that route?

**Comment by Deadreyo** [comment]
> Hum, would it be different to pass a handler as props to clear the input instead of using `onReset` handler?

Still requiring a new prop, didn't change anything.
Another way to go about this is to change the type of the `onchange` prop to be a function that takes string as input, and have a handler function for the state passed into it. Then from inside you can control what the state becomes.

File: components/atoms/TextInput/text-input.tsx [hunk 1]
```diff
@@ -35,17 +37,20 @@ const TextInput = ({
...
   pattern,
   errorMsg = ""
 }: TextInputProps) => {
   const inputRef = useRef<HTMLInputElement>(null);
   const handleResetInput = () => {
+    console.log({fieldRef});
     fieldRef ? fieldRef.current!.value = ""
       :
-      inputRef.current!.value = "";
+      onReset();
```
[the remark this answers is not in the document]
Comment (TAKANOME-DEV) [reply components/atoms/TextInput/text-input.tsx:51]: Yeah, that's right :+1:

[the remark this answers is not in the document]
Comment (TAKANOME-DEV) [reply components/atoms/TextInput/text-input.tsx:51]:
```suggestion
      onReset?.();
```

File: components/atoms/TextInput/text-input.tsx [hunk 2]
```diff
@@ -35,17 +37,20 @@ const TextInput = ({
...
   onChange,
+  onReset,
   required,
   fieldRef,
   pattern,
   errorMsg = ""
 }: TextInputProps) => {
   const inputRef = useRef<HTMLInputElement>(null);
   const handleResetInput = () => {
+    console.log({fieldRef});
```
Comment (brandonroberts) [inline components/atoms/TextInput/text-input.tsx:48]: nit: console log

```suggestion
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]
Merge conflicts should be resolved also

**Comment by TAKANOME-DEV** [comment]
> Another way to go about this is to change the type of the `onchange` prop to be a function that takes string as input, and have a handler function for the state passed into it. Then from inside you can control what the state becomes.
>
Can you provide an example?

**Review by brandonroberts — APPROVED** [review]

**Comment by Deadreyo** [comment]
> > Another way to go about this is to change the type of the `onchange` prop to be a function that takes string as input, and have a handler function for the state passed into it. Then from inside you can control what the state becomes.
> 
> Can you provide an example?

A simple example consisting of "component1" and "component2"
```ts
const component1 = () => {
const [state, setState] = useState("")

const handleChangeState = (value) => setState(value)

return (
 <component2 handleChange={handleChangeState} />
```
```ts
const component2 = ({ handleChangeState }) => {

const handleReset = () => handleChangeState("")
const handleChange = (e) => handleChangeState(e.target.value)
return (
<input onchange={handleChange} />
```
The difference is that the function in this example is a simple function that takes a string and changes the state, so the Reset can be done without needing another function. However the current implementation is that the function passed as prop takes an event, so can't do this. My suggestion is changing that.

**Comment by TAKANOME-DEV** [comment]
Thanks for the example @Deadreyo :+1:

**Review by Deadreyo — APPROVED** [review]
LGTM!

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**